### PR TITLE
Rename Account to AccountID, Tidy up jtx, add SHAMap::iterator

### PR DESCRIPTION
**Pull request by vinniefalco**
For the `AccountID` rename. Every line needs to be checked to make sure I didn't do something stupid like rename Account inside a SQL string. **NOTE** The first commit is the result of a search and replace while the second commit contains all the manual fixes.

### CODE REVIEW

File: src/ripple/app/main/Application.cpp [hunk 1]
```diff
@@ -1502,7 +1502,7 @@ static void addTxnSeqField ()
     }
 
     WriteLog (lsINFO, Application) << "Building new index";
-    session << "CREATE INDEX AcctTxIndex ON AccountTransactions(Account, LedgerSeq, TxnSeq, TransID);";
+    session << "CREATE INDEX AcctTxIndex ON AccountTransactions(AccountID, LedgerSeq, TxnSeq, TransID);";
```
Comment (nbougalis) [inline src/ripple/app/main/Application.cpp]: You're renaming an SQL column here.

Comment (vinniefalco) [inline src/ripple/app/main/Application.cpp]: good eyes

**Comment by nbougalis** [comment]
:+1:

**Comment by rec** [comment]
:+1:

**Pull request closed without merging**